datalake/src/test/java/com/sequenceiq/datalake/service/resize/recovery/ResizeRecoveryServiceTest.java

```diff
@@ -42,6 +54,12 @@ public class ResizeRecoveryServiceTest {
 
     private static final String USER_CRN = "crn:cdp:iam:us-west-1:cloudera:user:bob@cloudera.com";
 
+    private static final Set<DatalakeStatusEnum> FAILURE_STATES = Set.of(
+            PROVISIONING_FAILED, DELETE_FAILED, START_FAILED, STOP_FAILED, CLUSTER_AMBIGUOUS, CLUSTER_UNREACHABLE,
+            NODE_FAILURE, SYNC_FAILED, CERT_ROTATION_FAILED, CERT_RENEWAL_FAILED, DATALAKE_RESTORE_FAILED, RECOVERY_FAILED,
+            DATAHUB_REFRESH_FAILED
+    );
```

Comment:
when someone changes the set in the service class, test will be not valid anymore. you should not copy the whole set into test, but test the logic through valid methods.
you can use parameterized test to generated sdx cluster object with different failed states and check the logic against the sdx cluster object.

Reply:
Sorry Adam, I understand what you are saying but am confused about your advice. With my understanding I don't really see how we can maintain a global set of states like this that can be used both in testing and in the main class.
Previously I have been told not to make methods or fields public only for the case of testing, so I am confused as to how I can retain this information in both the test and the original class itself.

If the goal is to ensure that the same states used in the service class are used in the test, how can I do this in a way that does not violate the rule of fields/methods not being public purely for testing purposes?

Reply:
I agree with both the comments made.

Reply:
@sxxgrc
you can:
- made the set package private, in that case you can access it in test
- you can use public static method to return a set instead of a field
- i think you do not even need to go through all of the failed states during test, but pick one and test the logic for that state. My opinion is that unit test is to validate the business logic, not a static list/collection